addons/project/models/project.py

```diff
@@ -2071,6 +2107,27 @@ def action_open_parent_task(self):
             'context': self._context
         }
 
+    def action_subtask(self):
+        action = self.env["ir.actions.actions"]._for_xml_id("project.project_task_action_sub_task")
+
+        # display all subtasks of current task
+        action['domain'] = [('id', 'child_of', self.id), ('id', '!=', self.id)]
```

Comment:
@xavierbol , like this?: `<field name="domain">[('id', 'child_of', active_id), ('id', '!=', active_id)]</field>` but in that case we have id of project as `active_id` when we travel through the particular project(not the id of parent task). 

Thanks

Reply:
@xavierbol , Yes.. action will be available only in the task form.  but even though it will not pass the id of the task as active_id in subtask. ([see here](https://watch.screencastify.com/v/x3EBFZuZBNG6zh3MFtmY))
it might be a web issue. so for now, we should keep the domain in this method. 

Thanks

Reply:
I think the `active_id` used in the action is not really the one in the URL but the `self.id`. If you test by adding the domain in the XML definition of the action then it seems to work as you did in this method. 🙂

Reply:
@xavierbol , in my above comment I have added the video by adding the domain in XML, not from the method. Thanks

Reply:
@xavierbol , it will work when we direct open the sub-task from the `my Tasks` menu. but when we travel through the project it will create a problem see: https://watch.screencastify.com/v/q30cz8qt7E55TAsHgP2x

Thanks